<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Literature review-genomic selection in wheat breeding | Jing&#39;s Blog</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  
  
  
  
  <meta name="description" content="Literature overview Title: The value of early-stage phenotyping for wheat breeding in the age of genomic selection Author: Borrenpohl Daniel, Huang Mao, Olson Eric, Sneller Clay Institute: Department">
<meta property="og:type" content="article">
<meta property="og:title" content="Literature review-genomic selection in wheat breeding">
<meta property="og:url" content="https://jing-panda/Jing-panda.github.io/2020/12/09/Literature%20review/index.html">
<meta property="og:site_name" content="Jing&#39;s Blog">
<meta property="og:description" content="Literature overview Title: The value of early-stage phenotyping for wheat breeding in the age of genomic selection Author: Borrenpohl Daniel, Huang Mao, Olson Eric, Sneller Clay Institute: Department">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://i.loli.net/2020/12/09/bd4jHiBExAPF7D2.png">
<meta property="og:image" content="https://github.githubassets.com/images/icons/emoji/unicode/2b50.png?v8">
<meta property="og:image" content="https://i.loli.net/2020/12/09/lifXGI1mcEr7t8Y.png">
<meta property="og:image" content="https://i.loli.net/2020/12/09/2rJTE8zqICVg5Ht.png">
<meta property="article:published_time" content="2020-12-09T09:34:34.000Z">
<meta property="article:modified_time" content="2020-12-09T12:48:14.168Z">
<meta property="article:author" content="Jing Xu">
<meta property="article:tag" content="Forestgenetics, machinelearning, life, thoughts">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://i.loli.net/2020/12/09/bd4jHiBExAPF7D2.png">
  
    <link rel="alternate" href="/atom.xml" title="Jing&#39;s Blog" type="application/atom+xml">
  
  <link rel="icon" href="/css/images/favicon.ico">
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  <link href="https://fonts.googleapis.com/css?family=Open+Sans|Montserrat:700" rel="stylesheet" type="text/css">
  <link href="https://fonts.googleapis.com/css?family=Roboto:400,300,300italic,400italic" rel="stylesheet" type="text/css">
  <link href="//cdn.bootcss.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet">
  <style type="text/css">
    @font-face{font-family:futura-pt;src:url(https://use.typekit.net/af/9749f0/00000000000000000001008f/27/l?subset_id=2&fvd=n5) format("woff2");font-weight:500;font-style:normal;}
    @font-face{font-family:futura-pt;src:url(https://use.typekit.net/af/90cf9f/000000000000000000010091/27/l?subset_id=2&fvd=n7) format("woff2");font-weight:500;font-style:normal;}
    @font-face{font-family:futura-pt;src:url(https://use.typekit.net/af/8a5494/000000000000000000013365/27/l?subset_id=2&fvd=n4) format("woff2");font-weight:lighter;font-style:normal;}
    @font-face{font-family:futura-pt;src:url(https://use.typekit.net/af/d337d8/000000000000000000010095/27/l?subset_id=2&fvd=i4) format("woff2");font-weight:400;font-style:italic;}</style>
    
  <link rel="stylesheet" id="athemes-headings-fonts-css" href="//fonts.googleapis.com/css?family=Yanone+Kaffeesatz%3A200%2C300%2C400%2C700&amp;ver=4.6.1" type="text/css" media="all">

  <link rel="stylesheet" id="athemes-headings-fonts-css" href="//fonts.googleapis.com/css?family=Oswald%3A300%2C400%2C700&amp;ver=4.6.1" type="text/css" media="all">
  
<link rel="stylesheet" href="/css/style.css">


  
<script src="/js/jquery-3.1.1.min.js"></script>


  <!-- Bootstrap core CSS -->
  <link rel="stylesheet" href="/css/bootstrap.css" >
  <link rel="stylesheet" href="/css/fashion.css" >
  <link rel="stylesheet" href="/css/glyphs.css" >

<meta name="generator" content="Hexo 5.2.0">
<style>.github-emoji { position: relative; display: inline-block; width: 1.2em; min-height: 1.2em; overflow: hidden; vertical-align: top; color: transparent; }  .github-emoji > span { position: relative; z-index: 10; }  .github-emoji img, .github-emoji .fancybox { margin: 0 !important; padding: 0 !important; border: none !important; outline: none !important; text-decoration: none !important; user-select: none !important; cursor: auto !important; }  .github-emoji img { height: 1.2em !important; width: 1.2em !important; position: absolute !important; left: 50% !important; top: 50% !important; transform: translate(-50%, -50%) !important; user-select: none !important; cursor: auto !important; } .github-emoji-fallback { color: inherit; } .github-emoji-fallback img { opacity: 0 !important; }</style>
</head>



  <body data-spy="scroll" data-target="#toc" data-offset="50">


  
  
  <div class="site-header-image">
    <img id="originBg" width="100%" alt="Hike News" src="">
  </div>

  <div id="header-blur" class="site-header-image blur" style="position: absolute; top:0; height: 207px; min-height: 207px; min-width: 100%;">
    <img id="blurBg" width="100%" style="top: 96%" alt="Hike News" src="">
  </div>

  <script>
        var imgUrls = "css/images/pose01.jpg,https://source.unsplash.com/collection/954550/1920x1080,https://source.unsplash.com/collection/954550/1920x1081".split(",");
        var random = Math.floor((Math.random() * imgUrls.length ));
        if (imgUrls[random].startsWith('http') || imgUrls[random].indexOf('://') >= 0) {
          document.getElementById("originBg").src=imgUrls[random];
          document.getElementById("blurBg").src=imgUrls[random];
        } else {
          document.getElementById("originBg").src='/' + imgUrls[random];
          document.getElementById("blurBg").src='/' + imgUrls[random];
        }
    </script>




<header id="allheader" class="site-header" role="banner" 
   style="width: 100%; position: absolute; top:0; background: rgba(255,255,255,.8);"  >
  <div class="clearfix container">
      <div class="site-branding">

          <h1 class="site-title">
            
              <a href="/" title="Jing&#39;s Blog" rel="home"> Jing&#39;s Blog </a>
            
          </h1>
          
          
            <div class="site-description">Never stop learn</div>
          
            
          <nav id="main-navigation" class="main-navigation" role="navigation">
            <a class="nav-open">Menu</a>
            <a class="nav-close">Close</a>

            <div class="clearfix sf-menu">
              <ul id="main-nav" class="menu sf-js-enabled sf-arrows"  style="touch-action: pan-y;">
                    
                      <li class="menu-item menu-item-type-custom menu-item-object-custom menu-item-home menu-item-1663" linktext="/"> <a class="" href="/">Home</a> </li>
                    
                      <li class="menu-item menu-item-type-custom menu-item-object-custom menu-item-home menu-item-1663" linktext="archives"> <a class="" href="/archives">Archives</a> </li>
                    
                      <li class="menu-item menu-item-type-custom menu-item-object-custom menu-item-home menu-item-1663" linktext="about"> <a class="" href="/about">About</a> </li>
                    
              </ul>
            </div>
          </nav>

      </div>
  </div>
</header>


  <div id="container">
    <div id="wrap">
            
      <div id="content" class="outer">
        
          <section id="main" style="float:none;"><article id="post-Literature review" style="width: 66%; float:left;" class="article article-type-post" itemscope itemprop="blogPost" >
  <div id="articleInner" class="clearfix post-1016 post type-post status-publish format-standard has-post-thumbnail hentry category-template-2 category-uncategorized tag-codex tag-edge-case tag-featured-image tag-image tag-template">
    
    
      <header class="article-header">
        
  
    <h1 class="thumb" class="article-title" itemprop="name">
      Literature review-genomic selection in wheat breeding
    </h1>
  

      </header>
    
    <div class="article-meta">
      
	<a href="/2020/12/09/Literature%20review/" class="article-date">
	  <time datetime="2020-12-09T09:34:34.000Z" itemprop="datePublished">December 9, 2020</time>
	</a>

      
	<span class="ico-folder"></span>
    <a class="article-category-link" href="/categories/Literature-review/">Literature review</a>, <a class="article-category-link" href="/categories/Literature-review/genomic-selection/">genomic selection</a>
 
      
	<span id="busuanzi_container_page_pv">
	  本文总阅读量<span id="busuanzi_value_page_pv"></span>次
	</span>

    </div>
    <div class="article-entry" itemprop="articleBody">
      
        <h1>Literature overview</h1>
<p><strong>Title:</strong> The value of early-stage phenotyping for wheat breeding in the age of genomic selection</p>
<p><strong>Author:</strong> Borrenpohl Daniel, Huang Mao, Olson Eric, Sneller Clay</p>
<p><strong>Institute</strong>: Department of Horticulture and Crop Science, Ohio Agriculture Research and Development Center, The Ohio State University, USA</p>
<p><strong>Journal:</strong> <em>Theor Appl Genet</em></p>
<p><strong>Date:</strong> 01 June 2020</p>
<p><strong>DOI:</strong> <a target="_blank" rel="noopener" href="https://doi.org/10.1007/s00122-020-03613-0">https://doi.org/10.1007/s00122-020-03613-0</a></p>
<a id="more"></a>
<img src="https://i.loli.net/2020/12/09/bd4jHiBExAPF7D2.png" alt="image-20201209104311153">
<h1>Introduction</h1>
<h2 id="Genomic-selection"><strong>Genomic selection</strong></h2>
<p><strong>Genomic selection</strong> was first proposed by <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1461589/">Meuwissen et al. (2001)</a>.</p>
<p><strong>Theodorus Meuwissen</strong> is a professor at Norwegian University of Life Sciences, in the area of animal genetics.</p>
<p>Here is a video of his presentation about genomic selection:</p>
<iframe width="949" height="534" src="https://www.youtube.com/embed/xrsimLR48ww" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe>
<h3 id="Summary-from-presentation-of-Meuwissen">Summary from presentation of Meuwissen</h3>
<h4 id="Why-MAS-marker-assistant-selection-not-the-solution"><strong>Why MAS (marker assistant selection) not the solution</strong></h4>
<ul>
<li>
<p>Traits of interest are complex with many genes controlled, and affected by environment</p>
</li>
<li>
<p>Detection of QTL (quantitative traits loci) explain only small fraction of genetic variation</p>
</li>
</ul>
<h4 id="Three-breakthroughs-he-mentioned"><strong>Three breakthroughs he mentioned</strong></h4>
<ul>
<li>
<p>Detection of large number of SNP marker— Byproduct of sequencing efforts</p>
</li>
<li>
<p>SNP-chip genotyping technology — Affordable to genotype 50, 000 SNPs</p>
</li>
<li>
<p>Development of GS/ GP technology</p>
</li>
</ul>
<h4 id="Some-key-features-of-GS"><strong>Some key features of GS</strong></h4>
<ul>
<li>
<p>SNP effects are not tested for statistical significance — all SNP effects are used $\geqslant$ all genetic variance are addressed</p>
</li>
<li>
<p>Accurate GEBV for individuals without records — decouples elite breeding from recording</p>
<ul>
<li>Invasive recordings</li>
<li>Late in life records</li>
<li>Sex limited traits</li>
</ul>
</li>
<li>
<p>No pedigree recoding needed, because pedigree is replaced by marker based relationships</p>
</li>
</ul>
<h4 id="Statistic-model">Statistic model</h4>
<ul>
<li>
<p>linear regression: SNP-BLUP $\equiv$ GBLUP</p>
<p>$$<br>
y_i= \mu+X_{1x}* b_1 + X_{2i}* b_2 + … + X_{5000000} * b_{50000} + e_i<br>
$$</p>
<p>$$<br>
GEBV_j= X_{1j}* b_1 + X_{2j} * b_2 + … + X_{500000j}*b_{500000}<br>
$$</p>
<p><span class="github-emoji"><span>⭐</span><img src="https://github.githubassets.com/images/icons/emoji/unicode/2b50.png?v8" aria-hidden="true" onerror="this.parent.classList.add('github-emoji-fallback')"></span>all SNPs are same, and have the same constant variation, with mean of 0.</p>
</li>
<li>
<p>non-linear regression -Bayes A, Bayes B, Bayes C</p>
<p>$$<br>
y_i = I_1 * X_{i1}* b_1 + I_2 * X_{i2} * b_2 + … + I_{500000} * X_{i500000} * b_{500000} + e_i<br>
$$</p>
<ul>
<li>
<p>$I_j=0/1$ is an indicator whether SNP j has effect or not</p>
</li>
<li>
<p>Prior information is that fraction $\pi$ of SNPs have effect &amp; $(1-\pi)$ have no effect</p>
</li>
<li>
<p>Try to give extra weight to some SNPs and less to others</p>
</li>
</ul>
</li>
</ul>
<p>Simulation studies: GBLUP &lt; Bayes A &lt;Bayes C</p>
<p>Real data studies show:</p>
<ul>
<li>Differences small</li>
<li>often GBLUP is as good as BayesA/B</li>
</ul>
<p>Theoretically:</p>
<ul>
<li>If few and dense SNPs, BayesB is best</li>
<li>If many genes / not enough density: GBLUP best</li>
</ul>
<h4 id="Precision-breeding-Flint-Woolliams-2007">Precision breeding <a target="_blank" rel="noopener" href="https://royalsocietypublishing.org/doi/10.1098/rstb.2007.2171">(Flint &amp; Woolliams, 2007)</a></h4>
<ol>
<li>Accurate GEBV</li>
<li>Avoid deleterious side effects from breeding — select for broad breeding goal</li>
<li>Manage genetic variation — Realize precision breeding also into the future</li>
</ol>
<blockquote>
<p>In GS, a training population (TP) is formed if individuals who are phenotyped and genotyped with molecular markers. Those data are co-analyzed to build a prediction model.</p>
<p>The model can then be used to predict the trait value of unphenotyped individuals that <strong>are related to</strong> the TP and have been genotyped with the same markers.</p>
<p>The GS model calculates genomic estimated breeding values (GEBVs) for the unphenotyped individuals that can be used for selection in the same manner as phenotypic data.</p>
</blockquote>
<p><img src="https://i.loli.net/2020/12/09/lifXGI1mcEr7t8Y.png" alt="genomic selection.png"></p>
<p>link of image: <a target="_blank" rel="noopener" href="https://wiki.groenkennisnet.nl/display/TAB/Chapter+8.15.1%3A+Principle+of+genomic+selection">https://wiki.groenkennisnet.nl/display/TAB/Chapter+8.15.1%3A+Principle+of+genomic+selection</a></p>
<h2 id="Application-of-GS-in-plant-breeding">Application of GS in plant breeding</h2>
<h3 id="Population-improvement-phase">Population improvement phase</h3>
<p>The first application is for <strong><em>the population improvement phase of a program</em></strong> where recombination creates new genetic variation and <strong>testable progeny are generated</strong>. GS in this phase can increase <strong>the rate of genetic gain</strong> by <strong>reducing the duration of a recurrent selection breeding cycle</strong>.</p>
<h3 id="Product-development-phase-—-field-testing">Product development phase — field testing</h3>
<p>GS can also be applied to <strong><em>the product development phase of breeding</em></strong> where progeny are evaluated for performance in a stage-gate process. GS could enhance field trial selection by integrating genotypic and phenotypic data as opposed to advancing lines using phenotypic data alone.</p>
<h2 id="Obejectives">Obejectives</h2>
<ol>
<li>Compare the prediction accuracy of GS and phenotyping in stage-1 and stage-2 field evaluations</li>
<li>Assess the value of stage-1 phenotyping for advancing lines to stage-2 testing</li>
</ol>
<h1>Materials and methods</h1>
<p><strong>Training population</strong>: <strong>1769</strong> wheat breeding lines (genotyped +phenotypes)</p>
<p><strong>Traits</strong>: yield, test weight, Fusarium head blight resistance, heading date, and height.</p>
<p><img src="https://i.loli.net/2020/12/09/2rJTE8zqICVg5Ht.png" alt="Fig. 3"></p>
<p><strong>SNP discovery</strong>: 400 randomly selected lines was first used for single nucleotide polymorphism (SNP) discovery by comparing their sequences to the wheat reference genome in the TASSEL-GBS pipeline (Glaubitz et al. <a target="_blank" rel="noopener" href="https://link.springer.com/article/10.1007/s00122-020-03613-0#ref-CR15">2014</a>). In total, ~ <strong>190,000</strong> SNPs were detected.</p>
<p><strong>SNP cleaning</strong>: Only SNPs with less than 20% missing data and greater than 0.05 minor allele frequency were retained. Missing marker scores were imputed using an <strong>expectation–maximization algorithm</strong> in R package <strong>rrBLUP</strong> (v. 4.6) using the “<strong>A.mat</strong>” function (Endelman <a target="_blank" rel="noopener" href="https://link.springer.com/article/10.1007/s00122-020-03613-0#ref-CR12">2011</a>). To ensure even marker genome coverage and reduce marker redundancy, <strong>a SNP tagging procedure using a threshold of <em>r</em> = 0.8</strong> was conducted to identify the most informative markers (Rinaldo et al. <a target="_blank" rel="noopener" href="https://link.springer.com/article/10.1007/s00122-020-03613-0#ref-CR41">2005</a>; Huang et al. <a target="_blank" rel="noopener" href="https://link.springer.com/article/10.1007/s00122-020-03613-0#ref-CR24">2016</a>). After this tagging procedure, a total of <strong>12,037 high-quality and evenly distributed SNPs</strong> were used for analysis.</p>
<p><strong>Population structure:</strong> Population structure among all 1769 lines was assessed using principal component analysis of the marker data in the <strong>R stats package</strong> (v. 3.5.3) using the “<strong>prcomp</strong>” function (R Core Development Team <a target="_blank" rel="noopener" href="https://link.springer.com/article/10.1007/s00122-020-03613-0#ref-CR49">2019</a>).</p>
<p><strong>Modeling</strong>: GS models were built using R package <strong>rrBLUP</strong> (v. 4.6) to estimate marker effects for each trait using <strong>ridge regression BLUP (RR-BLUP)</strong> using the “<strong>mixed.solve</strong>” function (Endelman, <a target="_blank" rel="noopener" href="https://link.springer.com/article/10.1007/s00122-020-03613-0#ref-CR12">2011</a>). RR-BLUP was chosen primarily for computational efficiency because Huang et al. (<a target="_blank" rel="noopener" href="https://link.springer.com/article/10.1007/s00122-020-03613-0#ref-CR24">2016</a>) found that RR-BLUP had comparable accuracy to other GS models for all wheat traits in the Ohio State University breeding program.</p>
<h1>Conclusion</h1>
<blockquote>
<p>In conclusion, GS is proving to be a valuable tool for improving the efficiency of the population improvement and product development phases of breeding field trials. Introducing GS into a program can have a dramatic benefit to a breeding program. But applying GS requires a thorough assessment of the impact of GS on genetic gain, cost per unit of gain, allocation of field-testing resources, and operational considerations in all phases of breeding phases.</p>
</blockquote>

      
    </div>
    <footer class="entry-meta entry-footer">
      
	<span class="ico-folder"></span>
    <a class="article-category-link" href="/categories/Literature-review/">Literature review</a>, <a class="article-category-link" href="/categories/Literature-review/genomic-selection/">genomic selection</a>

      
      
        
	<div id="comment">
		<!-- 来必力City版安装代码 -->
		<div id="lv-container" data-id="city" data-uid="MTAyMC8yOTQ4MS82MDQ5">
		<script type="text/javascript">
		   (function(d, s) {
		       var j, e = d.getElementsByTagName(s)[0];

		       if (typeof LivereTower === 'function') { return; }

		       j = d.createElement(s);
		       j.src = 'https://cdn-city.livere.com/js/embed.dist.js';
		       j.async = true;

		       e.parentNode.insertBefore(j, e);
		   })(document, 'script');
		</script>
		<noscript>为正常使用来必力评论功能请激活JavaScript</noscript>
		</div>
		<!-- City版安装代码已完成 -->
	</div>



      
    </footer>
    <hr class="entry-footer-hr">
  </div>
  
    
<nav id="article-nav">
  
  
    <a href="/2020/12/08/Dansk%20ord%20hver%20dag2/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">Dansk ord hver dag 2</div>
    </a>
  
</nav>

  
</article>

<!-- Table of Contents -->

  <aside id="sidebar">
    <div id="toc" class="toc-article">
    <strong class="toc-title">Contents</strong>
    
      <ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link"><span class="nav-number">1.</span> <span class="nav-text">Literature overview</span></a></li><li class="nav-item nav-level-1"><a class="nav-link"><span class="nav-number">2.</span> <span class="nav-text">Introduction</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genomic-selection"><span class="nav-number">2.1.</span> <span class="nav-text">Genomic selection</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Summary-from-presentation-of-Meuwissen"><span class="nav-number">2.1.1.</span> <span class="nav-text">Summary from presentation of Meuwissen</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Why-MAS-marker-assistant-selection-not-the-solution"><span class="nav-number">2.1.1.1.</span> <span class="nav-text">Why MAS (marker assistant selection) not the solution</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Three-breakthroughs-he-mentioned"><span class="nav-number">2.1.1.2.</span> <span class="nav-text">Three breakthroughs he mentioned</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Some-key-features-of-GS"><span class="nav-number">2.1.1.3.</span> <span class="nav-text">Some key features of GS</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Statistic-model"><span class="nav-number">2.1.1.4.</span> <span class="nav-text">Statistic model</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Precision-breeding-Flint-Woolliams-2007"><span class="nav-number">2.1.1.5.</span> <span class="nav-text">Precision breeding (Flint &amp; Woolliams, 2007)</span></a></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Application-of-GS-in-plant-breeding"><span class="nav-number">2.2.</span> <span class="nav-text">Application of GS in plant breeding</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Population-improvement-phase"><span class="nav-number">2.2.1.</span> <span class="nav-text">Population improvement phase</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Product-development-phase-%E2%80%94-field-testing"><span class="nav-number">2.2.2.</span> <span class="nav-text">Product development phase — field testing</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Obejectives"><span class="nav-number">2.3.</span> <span class="nav-text">Obejectives</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link"><span class="nav-number">3.</span> <span class="nav-text">Materials and methods</span></a></li><li class="nav-item nav-level-1"><a class="nav-link"><span class="nav-number">4.</span> <span class="nav-text">Conclusion</span></a></li></ol>
    
    </div>
  </aside>
</section>
        
      </div>

    </div>
    <!-- <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
</nav> -->
    <footer id="footer" class="site-footer">
  

  <div class="clearfix container">
      <div class="site-info">
	      &copy; 2020 Jing&#39;s Blog All Rights Reserved.
        
            <span id="busuanzi_container_site_uv">
              本站访客数<span id="busuanzi_value_site_uv"></span>人次  
              本站总访问量<span id="busuanzi_value_site_pv"></span>次
            </span>
          
      </div>
      <div class="site-credit">
        Theme by <a href="https://github.com/iTimeTraveler/hexo-theme-hipaper" target="_blank">hipaper</a>
      </div>
  </div>
</footer>


<!-- min height -->

<script>
    var wrapdiv = document.getElementById("wrap");
    var contentdiv = document.getElementById("content");

    wrapdiv.style.minHeight = document.body.offsetHeight - document.getElementById("allheader").offsetHeight - document.getElementById("footer").offsetHeight + "px";
    contentdiv.style.minHeight = document.body.offsetHeight - document.getElementById("allheader").offsetHeight - document.getElementById("footer").offsetHeight + "px";


    <!-- headerblur min height -->
    
      var headerblur = document.getElementById("header-blur");
      headerblur.style.minHeight = window.getComputedStyle(document.getElementById("allheader"), null).height;
    
    
</script>

<script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    
<div style="display: none;">
  <script src="https://s11.cnzz.com/z_stat.php?id=1260716016&web_id=1260716016" language="JavaScript"></script>
</div>

<!-- mathjax config similar to math.stackexchange -->

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for(i=0; i < all.length; i += 1) {
            all[i].SourceElement().parentNode.className += ' has-jax';
        }
    });
</script>

<script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>


<script src="/js/bootstrap.js"></script>


<script src="/js/main.js"></script>









	<script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js">
	</script>






  </div>

  <a id="rocket" href="#top" class=""></a>
  <script type="text/javascript" src="/js/totop.js" async=""></script>
</body>
</html>
